src/game/sprite.ts

```diff
@@ -0,0 +1,63 @@
+import Resources from './resources';
+
+export default class Sprite {
+  private _index: number;
+  url: string;
+  pos: [number, number];
+  size: [number, number];
+  speed?: number;
+  frames?: [number, number];
+  dir?: 'horizontal' | 'vertical';
+  once?: boolean;
+  done?: boolean;
+
+  constructor(url: string, pos: [number, number], size: [number, number], speed?: number, frames?: [number, number], dir?: 'horizontal' | 'vertical', once?: boolean) {
+    this._index = 0;
+    this.url = url;
+    this.pos = pos;
+    this.size = size;
+    this.speed = speed === undefined ? 0 : speed;
```

Comment:
🧐
Потребуется дополнительный [babel-плагин](https://babeljs.io/docs/en/babel-plugin-proposal-nullish-coalescing-operator), но такой синтаксис уже активно юзается
```suggestion
    this.speed = speed ?? 0;
```